lib/horde/supervisor.ex

```diff
@@ -155,49 +165,38 @@ defmodule Horde.Supervisor do
   end
 
   @doc false
-  def handle_call({:delete_child, child_id}, _from, state) do
-    with {:ok, supervisor_pid} <- which_supervisor(child_id, state),
-         :ok <- Supervisor.delete_child(supervisor_pid, child_id) do
-      GenServer.cast(
-        state.processes_pid,
-        {:operation, {@crdt, :remove, [child_id]}}
-      )
-
-      {:reply, :ok, state}
-    else
-      {:error, error} -> {:reply, {:error, error}, state}
-    end
-  end
-
-  @doc false
-  def handle_call({:restart_child, child_id}, _from, state) do
-    case which_supervisor(child_id, state) do
-      {:ok, supervisor_pid} ->
-        {:reply, Supervisor.restart_child(supervisor_pid, child_id), state}
-
-      error ->
-        {:reply, error, state}
-    end
-  end
+  def handle_call({:terminate_child, child_id} = msg, from, %{node_id: this_node_id} = state) do
+    case Map.get(state.processes, child_id) do
+      {^this_node_id, _child_spec} ->
+        reply =
+          DynamicSupervisor.terminate_child(
+            state.supervisor_pid,
+            Process.whereis(Horde.ProcessSupervisor.name(this_node_id, child_id))
+          )
 
-  @doc false
-  def handle_call({:terminate_child, child_id}, _from, state) do
-    case which_supervisor(child_id, state) do
-      {:ok, supervisor_pid} ->
-        {:reply, Supervisor.terminate_child(supervisor_pid, child_id), state}
+        new_state = %{state | processes: Map.delete(state.processes, child_id)}
+        :ok = GenServer.cast(state.processes_pid, {:operation, {@crdt, :remove, [child_id]}})
+        {:reply, reply, new_state}
 
-      error ->
-        {:reply, error, state}
+      {other_node, _child_spec} ->
+        proxy_to_node(other_node, msg, from, state)
     end
   end
 
   @doc false
   def handle_call({:start_child, _child_spec}, _from, %{shutting_down: true} = state),
-    do: {:reply, {:error, "shutting down"}, state}
+    do: {:reply, {:error, {:shutting_down, "this node is shutting down. please retry."}}, state}
```

Comment:
`please retry elsewhere`? if a node is shutting down, folks shouldn't "just retry" :P

Reply:
Removed retry part, we can't know why a node is shutting down.